### PR TITLE
Add SoX 14.4.2

### PR DESCRIPTION
**Pull request by AdrianArroyoCalle**

### CODE REVIEW

**Comment by waddlesplash** [comment]
@AdrianArroyoCalle Please fix the trailing whitespace.

File: media-sound/sox/sox-14.4.2.recipe [hunk 1]
```diff
@@ -0,0 +1,80 @@
+SUMMARY="A command line utility to do audio format conversions"
+DESCRIPTION="
+SoX is a cross-platform (Windows, Linux, MacOS X, etc.) command line utility \
+that can convert various formats of computer audio files in to other formats. \
+It can also apply various effects to these sound files, and, as an added \
+bonus, SoX can play and record audio files on most platforms 
+"
```
Comment (waddlesplash) [inline media-sound/sox/sox-14.4.2.recipe]: Put this on previous line, add period.

File: media-sound/sox/sox-14.4.2.recipe [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+HOMEPAGE="http://sox.sourceforge.net"
+SOURCE_URI="http://sourceforge.net/projects/sox/files/sox/14.4.2/sox-14.4.2.tar.bz2"
+CHECKSUM_SHA256="81a6956d4330e75b5827316e44ae381e6f1e8928003c6aa45896da9041ea149c"
+LICENSE="GNU GPL v2"
+COPYRIGHT="1991-2015 Lance Norskog"
```
Comment (waddlesplash) [inline media-sound/sox/sox-14.4.2.recipe]: Reverse the order of this and LICENSE and put them below HOMEPAGE

File: media-sound/sox/sox-14.4.2.recipe [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+SECONDARY_ARCHITECTURES="x86_gcc2 x86"
+
+PROVIDES="
+	sox$secondaryArchSuffix = $portVersion compat >= 1
+	cmd:sox$secondaryArchSuffix = $portVersion
+	cmd:play$secondaryArchSuffix = $portVersion
+	cmd:rec$secondaryArchSuffix = $portVersion
+	cmd:soxi$secondaryArchSuffix = $portVersion
+	lib:libsox$secondaryArchSuffix = 0.0.0 compat >= 0
+"
```
Comment (waddlesplash) [inline media-sound/sox/sox-14.4.2.recipe]: Indent this with one tab. (Same below.)

File: media-sound/sox/sox-14.4.2.recipe [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
```
Comment (waddlesplash) [inline media-sound/sox/sox-14.4.2.recipe]: Delete this blank line

File: media-sound/sox/sox-14.4.2.recipe [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libflac$secondaryArchSuffix
+	devel:libvorbis$secondaryArchSuffix
+	devel:libogg$secondaryArchSuffix
+	devel:libmp3lame$secondaryArchSuffix
+	devel:libsndfile$secondaryArchSuffix
+	devel:libpng$secondaryArchSuffix
+"
+
```
Comment (waddlesplash) [inline media-sound/sox/sox-14.4.2.recipe]: And this one.

File: media-sound/sox/sox-14.4.2.recipe [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:autoconf
+"
+
+PATCHES="sox-14.4.2.patchset"
+
+SOURCE_DIR="sox-14.4.2"
```
Comment (waddlesplash) [inline media-sound/sox/sox-14.4.2.recipe]: Goes below CHECKSUM.

File: media-sound/sox/sox-14.4.2.recipe [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+"
+
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:autoconf
+"
+
+PATCHES="sox-14.4.2.patchset"
```
Comment (waddlesplash) [inline media-sound/sox/sox-14.4.2.recipe]: Goes below SOURCE_DIR.

File: media-sound/sox/sox-14.4.2.recipe [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+	fixPkgconfig
+	
+	packageEntries devel $developDir
+}
+
+PROVIDES_devel="
+	sox${secondaryArchSuffix}_devel = $portVersion
+	devel:libsox$secondaryArchSuffix = 0.0.0 compat >= 0
+"
+
```
Comment (waddlesplash) [inline media-sound/sox/sox-14.4.2.recipe]: Delete this blank line.

File: media-sound/sox/sox-14.4.2.recipe [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+REQUIRES_devel="
+	sox${secondaryArchSuffix} == $portVersion base
+"
```
Comment (waddlesplash) [inline media-sound/sox/sox-14.4.2.recipe]: Put PROVIDES_devel/REQUIRES_devel between PROVIDES/REQUIRES and BUILD_REQUIRES/PREREQUIRES.

**Comment by humdingerb** [comment]
See https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines for the correct ordering of the blocks.

**Comment by AdrianArroyoCalle** [comment]
Sorry to ask, but where's the trailing whitespace

**Comment by AdrianArroyoCalle** [comment]
Ah, Ok, I found the trailing whitespace just right now

**Comment by waddlesplash** [comment]
@AdrianArroyoCalle Build is still failing. You can enable automatic trailing whitespace stripping your editor (Pe has it in Preferences -> Files).

**Comment by humdingerb** [comment]
> It seems that you have updated the guidelines recently. There are lots of recipes with a wrong section order.

Right, the wiki is being kept uptodate. Feel free to help re-ordering recipes, though that's not a priority. Still, new recipes should conform.

**Comment by AdrianArroyoCalle** [comment]
Checks have passed now

**Comment by waddlesplash** [comment]
Thanks!

**Pull request merged**